0101-nix formatting.md

```diff
@@ -3,63 +3,546 @@ feature: nix_formatting
 start-date: 2021-08-17
 author: Raphael Megzari
 co-authors: (find a buddy later to help out with the RFC)
-shepherd-team: Timothy DeHerrera (nrdxp), 0x4A6F, Silvan Mosberger (infinisil)
-shepherd-leader: Jonas Chevalier (zimbatm)
+shepherd-team: 0x4A6F, Silvan Mosberger (infinisil), piegames, Thomas Bereknyei (tomberek)
+shepherd-leader: Silvan Mosberger (infinisil)
 related-issues: (will contain links to implementation PRs)
 ---
 
 # Summary
 
 [summary]: #summary
 
-Decide on a recommended automated formatter for nix files in nixpkgs.
+Decide on basic Nix style guidelines and pick a default code formatter for Nix code. 
+Format Nixpkgs using that formatter and enforce it using CI.
 
-# Motivation
+## Motivation
 
 [motivation]: #motivation
 
-Prevent debate around how things should be formatted.
+TODO: Currently, there is no authoritative style guide for Nix code in Nixpkgs.
+The currenty style has evolved organically over time, leading to inconsistencies both across files and language features.
 
-# Detailed design
+- We want to prevent debate around how things should be formatted.
+- We want to make it as easy as possible for contributors (especially new ones) to make changes without having to worry about formatting.
+- Conversely, reviewers should not be bothered with ill-formatted contributions.
+- We want a unified Nix code style that's consistent with itself, easily readable, accessible and results in readable diffs.
+
+## Detailed design
 
 [design]: #detailed-design
 
-The implementation of this RFC should include several parts
+There are four main parts to this RFC:
+- Establish basic formatting guidelines
+- Pick a formatter implementation
+- Migrate Nixpkgs to the format and enforce it in CI
```

Comment:
```suggestion
```

I feel like these two shouldn't be part of the RFC but rather its implementation.

As @infinisil said, as long as the tool adheres to the style decided here, it doesn't matter which one you use. I'd just leave this open or at most give a recommendation based on experience in creating this RFC. Obviously said recommendation would be the most sensible to implement at first but if it later turns out that some other tool would be better suited, you could start using that without having to create a new RFC as it's an objective decision.

What really need to be decided through the RFC process are the style and rules as they're subjective and therefore need a community decision.